File: LimDaeYoung/be-onboarding-project/api/common-api/src/main/java/com/survey/api/entity/SurveyOptionEntity.java
package com.survey.api.entity;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity(name="surveyOption")
public class SurveyOptionEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String optionName;
    private int optionOrder;
    private String useYn;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime regDtm;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "itemId")
    private SurveyItemEntity surveyItem; // 설문 항목 참조

    public SurveyOptionEntity(String optionName, int optionOrder, String useYn, SurveyItemEntity surveyItem) {
        this.optionName = optionName;
        this.optionOrder = optionOrder;
        this.useYn = useYn;
        this.surveyItem = surveyItem;
    }

    public SurveyOptionEntity(Long id,String optionName, int optionOrder, String useYn, SurveyItemEntity surveyItem) {
        this.id = id;
        this.optionName = optionName;
        this.optionOrder = optionOrder;
        this.useYn = useYn;
        this.surveyItem = surveyItem;
    }

    public SurveyOptionEntity(Long id) {
        this.id = id;
    }
}

File: survey-data/src/main/kotlin/com/ic/surveydata/entity/SurveyOptionEntity.kt
package com.ic.surveydata.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity
@Table(name = "survey_option")
data class SurveyOptionEntity(
    @Id
    @Column(name = "id", nullable = false, unique = true)
    val id: String,
    @Column(nullable = false, unique = false)
    val name: String,
    @ManyToOne
    @JoinColumn(name = "survey_item_id")
    val surveyItemEntity: SurveyItemEntity? = null,
    @Column(name = "created_at", nullable = false, unique = false)
    val createdAt: LocalDateTime = LocalDateTime.now(),
)
